spark/src/main/java/org/opensearch/sql/spark/execution/session/SessionModel.java

```diff
@@ -0,0 +1,143 @@
+        .field(SESSION_TYPE, sessionType.getSessionType())
+        .field(SESSION_ID, sessionId.getSessionId())
+        .field(SESSION_STATE, sessionState.getSessionState())
+        .field(DATASOURCE_NAME, datasourceName)
+        .field(APPLICATION_ID, applicationId)
+        .field(JOB_ID, jobId)
+        .field(LAST_UPDATE_TIME, lastUpdateTime)
+        .field(ERROR, error)
+        .endObject();
+    return builder;
+  }
+
+  public static SessionModel of(SessionModel copy, long seqNo, long primaryTerm) {
+    return builder()
+        .version(copy.version)
+        .sessionType(copy.sessionType)
+        .sessionId(new SessionId(copy.sessionId.getSessionId()))
+        .sessionState(copy.sessionState)
+        .datasourceName(copy.datasourceName)
+        .seqNo(seqNo)
+        .primaryTerm(primaryTerm)
+        .build();
+  }
+
+  @SneakyThrows
+  public static SessionModel fromXContent(XContentParser parser, long seqNo, long primaryTerm) {
+    SessionModelBuilder builder = new SessionModelBuilder();
+    XContentParserUtils.ensureExpectedToken(
+        XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+    while (!XContentParser.Token.END_OBJECT.equals(parser.nextToken())) {
+      String fieldName = parser.currentName();
+      parser.nextToken();
+      switch (fieldName) {
+        case VERSION:
+          builder.version(parser.text());
+          break;
+        case SESSION_TYPE:
+          builder.sessionType(SessionType.fromString(parser.text()));
+          break;
+        case SESSION_ID:
+          builder.sessionId(new SessionId(parser.text()));
+          break;
+        case SESSION_STATE:
+          builder.sessionState(SessionState.fromString(parser.text()));
+          break;
+        case DATASOURCE_NAME:
+          builder.datasourceName(parser.text());
+          break;
+        case ERROR:
+          builder.error(parser.text());
+          break;
+        case APPLICATION_ID:
+          builder.applicationId(parser.text());
+          break;
+        case JOB_ID:
+          builder.jobId(parser.text());
+          break;
+        case LAST_UPDATE_TIME:
+          builder.lastUpdateTime(parser.longValue());
+          break;
+        case TYPE:
+          // do nothing.
+          break;
+      }
+    }
+    builder.seqNo(seqNo);
+    builder.primaryTerm(primaryTerm);
+    return builder.build();
+  }
+
+  public static SessionModel initInteractiveSession(
+      String applicationId, String jobId, SessionId sid, String datasourceName) {
+    return builder()
+        .version("1.0")
+        .sessionType(INTERACTIVE)
+        .sessionId(sid)
+        .sessionState(NOT_STARTED)
+        .datasourceName(datasourceName)
+        .applicationId(applicationId)
+        .jobId(jobId)
+        .error(UNKNOWN)
+        .lastUpdateTime(System.currentTimeMillis())
+        .seqNo(SequenceNumbers.UNASSIGNED_SEQ_NO)
+        .primaryTerm(SequenceNumbers.UNASSIGNED_PRIMARY_TERM)
```

Comment:
so these unassigned numbers can ensure version conflict happen when create session in the beginning?

Reply:
yes.